src/App.tsx

```diff
@@ -1,14 +1,76 @@
 /* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
+import React, { useEffect, useState } from 'react';
+
+import { getTodos } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [query, setQuery] = useState('');
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.ALL);
+
+  useEffect(() => {
+    getTodos()
+      .then((todosFromServer) => {
+        setTodos(todosFromServer);
```

Comment:
It's a good practice to set 'isLoading' to false in the 'finally' block to ensure it's executed whether the promise is fulfilled or rejected.